cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -112,11 +119,46 @@ public void configure(Map<String, ?> configs) {
 
     _cruiseControlMetricsTopic = reporterConfig.getString(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_CONFIG);
     _reportingIntervalMs = reporterConfig.getLong(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_REPORTING_INTERVAL_MS_CONFIG);
+
+    if(reporterConfig.getBoolean(CruiseControlMetricsReporterConfig.CRUISE_CONTROL_METRICS_TOPIC_AUTO_CREATE_CONFIG)) {
+      _newTopic = createNewTopicFromReporterConfig(reporterConfig);
+      final Properties adminProperties = new Properties();
+      adminProperties.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, producerProps.get(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG));
+      adminProperties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, producerProps.get(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG));
```

Comment:
If the security protocol is `SSL`, I suspect that the `AdminClient` properties set here will be missing a number of required configs. (please see a similar patch in `migrate_to_kafka_2_0` branch of Cruise Control: https://github.com/linkedin/cruise-control/commit/2d7b519d806a7a2228a2b19fdedf6a457e6053d8)

Reply:
The producer already not set all this SSL properties. Does it currently work with SSL?

Reply:
-- I agree that it is not clear from the code -- I'll add pointers/comments to make it more clear. 
The [documentation on configurations](https://github.com/linkedin/cruise-control/wiki/Configurations) indicate that `CruiseControlMetricsReporter takes all the configurations for vanilla KafkaProducer with a prefix of "cruise.control.metrics.reporter."`.